src/Compilers/CSharp/Portable/Symbols/Source/SourceAssemblySymbol.cs

```diff
@@ -1428,34 +1470,41 @@ private void LoadAndValidateNetModuleAttributes(ref CustomAttributesBag<CSharpAt
 
                 if (Interlocked.CompareExchange(ref lazyNetModuleAttributesBag, netModuleAttributesBag, null) == null)
                 {
-                    this.AddDeclarationDiagnostics(diagnostics);
+                    if (!noCache)
+                    {
+                        this.AddDeclarationDiagnostics(diagnostics);
+                    }
                 }
 
                 diagnostics.Free();
-            }
 
-            Debug.Assert(lazyNetModuleAttributesBag.IsSealed);
+                Debug.Assert(lazyNetModuleAttributesBag.IsSealed);
+            }
         }
 
-        private void EnsureNetModuleAttributesAreBound()
```

Comment:
:memo: I just inlined `EnsureNetModuleAttributesAreBound` to the single callsite